### PR TITLE
[FEATURE] Add command for document delete

### PR DESCRIPTION
**Pull request by beatrycze-volk**

### CODE REVIEW

File: Configuration/Commands.php [hunk 1]
```diff
@@ -21,6 +21,9 @@
  * https://docs.typo3.org/c/typo3/cms-core/master/en-us/Changelog/10.3/Deprecation-89139-ConsoleCommandsConfigurationFormatCommandsPhp.html
  */
 return [
+    'kitodo:delete' => [
```
Comment (csidirop) [inline Configuration/Commands.php]: I believe this addition has no affect any more. (See comment from Alexander or here https://docs.typo3.org/c/typo3/cms-core/main/en-us/Changelog/10.3/Deprecation-89139-ConsoleCommandsConfigurationFormatCommandsPhp.html)

In fact we can remove all entries and are still able to call the commands.

**Comment by csidirop** [comment]
Was going to test your PR but run into an error, where the registration of the command in /Configuration/Commands.php causes errors in other commands:

```
./vendor/bin/typo3 kitodo:index -d https://digi.bib.uni-mannheim.de/fileadmin/digi/445662204/445662204.xml -p 3 -s dlf

Uncaught TYPO3 Exception Argument 1 passed to Kitodo\Dlf\Command\BaseCommand::__construct() must be an instance of Kitodo\Dlf\Domain\Repository\CollectionRepository, string given, called in /var/www/typo3/public/typo3/sysext/core/Classes/Utility/GeneralUtility.php on line 3492
thrown in file /var/www/typo3/public/typo3conf/ext/dlf/Classes/Command/BaseCommand.php
in line 92

```

Reversing the changes in Commands.php fixes the error. Can you reproduce or explain this behaviour?

File: Configuration/Commands.php (hunk 1 above)
Comment (csidirop) [reply Configuration/Commands.php]: I tried registering the command in `Configuration/Services.yaml` (https://docs.typo3.org/m/typo3/reference-coreapi/10.4/en-us/Configuration/Yaml/ServicesYaml.html) but it did not worked:

````
                                           
  Command "kitodo:delete" is not defined.  
                                           
  Did you mean one of these?               
      kitodo:harvest                       
      kitodo:index                         
      kitodo:reindex                                                                 

````

**Comment by beatrycze-volk** [comment]
I have fixed command registration according your comment and then tested everything again. The command still works for me. I couldn't reproduce your error. But as I have used more different documents to test I was able to find some other problems and fix them :)

Here are my test data and the results from the console:

`ddev typo3 kitodo:index --doc https://digital.slub-dresden.de/data/kitodo/NamezuGeS_187317411X_0002/NamezuGeS_187317411X_0002_mets.xml --pid 2 --solr 1 --verbose`

index4324

`ddev typo3 kitodo:delete --doc https://digital.slub-dresden.de/data/kitodo/NamezuGeS_187317411X_0002/NamezuGeS_187317411X_0002_mets.xml --pid 2 --solr 1 --verbose`

delete4324

**Review by csidirop — APPROVED** [review]
Working now for me too. 

But the entries still remain in TYPO3:
grafik

**Comment by beatrycze-volk** [comment]
Are you getting this message in console: 
`'Deleted xxxx ("https://.../mets.xml") on PID x.'` ?

**Comment by csidirop** [comment]
> Are you getting this message in console: `'Deleted xxxx ("https://.../mets.xml") on PID x.'` ?

Yes. Here my output:

````
root@localhost:/var/www/typo3# ./vendor/bin/typo3 kitodo:index --doc https://digital.slub-dresden.de/data/kitodo/NamezuGeS_187317411X_0002/NamezuGeS_187317411X_0002_mets.xml --pid 3 --solr dlf --verbose

Index single document into database and Solr.
=============================================

Indexing  ("https://digital.slub-dresden.de/data/kitodo/NamezuGeS_187317411X_0002/NamezuGeS_187317411X_0002_mets.xml") on PID 3 and Solr core 1.
------------------------------------------------------------------------------------------------------------------------------------------------


 [OK] All done!

````


````

root@localhost:/var/www/typo3# ./vendor/bin/typo3 kitodo:delete --doc https://digital.slub-dresden.de/data/kitodo/NamezuGeS_187317411X_0002/NamezuGeS_187317411X_0002_mets.xml --pid 3 --solr dlf --verbose

Delete single document from database and Solr.
==============================================

Deleting 2 ("https://digital.slub-dresden.de/data/kitodo/NamezuGeS_187317411X_0002/NamezuGeS_187317411X_0002_mets.xml") on PID 3.
---------------------------------------------------------------------------------------------------------------------------------


 [OK] Deleted 2
      ("https://digital.slub-dresden.de/data/kitodo/NamezuGeS_187317411X_0002/NamezuGeS_187317411X_0002_mets.xml") on
      PID 3.


Deleting https://digital.slub-dresden.de/data/kitodo/NamezuGeS_187317411X_0002/NamezuGeS_187317411X_0002_mets.xml on Solr core 1.
---------------------------------------------------------------------------------------------------------------------------------


 [OK] Deleted https://digital.slub-dresden.de/data/kitodo/NamezuGeS_187317411X_0002/NamezuGeS_187317411X_0002_mets.xml
      on Solr core 1.



 [OK] All done!

````

And here what it looks like in a clean installation after the `index` and after the `delete`:


index-delete


Solr query `*.*` returns:

```
{
  "responseHeader":{
    "status":0,
    "QTime":0,
    "params":{
      "q":"*:*",
      "indent":"true",
      "q.op":"OR",
      "_":"1706605607068"}},
  "response":{"numFound":1,"start":0,"numFoundExact":true,"docs":[
      {
        "id":"1uuid-2e18fc84-cf49-467a-a3f3-861335e65892",
        "uid":1,
        "partof":0,
        "root":0,
        "sid":"uuid-2e18fc84-cf49-467a-a3f3-861335e65892",
        "type":"multivolume_work",
        "collection":["Stadtarchiv Pirna"],
        "fulltext":"",
        "page":1,
        "thumbnail":"",
        "toplevel":true,
        "title":"Namensverzeichnisse zu den Geburtsregistern des Standesamtes Pirna",
        "date":"1876",
        "record_id":"oai:de:slub-dresden:db:id-1873169833",
        "purl":"http://digital.slub-dresden.de/id1873169833",
        "location":"https://digital.slub-dresden.de/data/kitodo/NamezudeG_1873169833/NamezudeG_1873169833_mets.xml",
        "urn":"urn:nbn:de:bsz:14-db-id18731698335",
        "title_tsi":["Namensverzeichnisse zu den Geburtsregistern des Standesamtes Pirna"],
        "title_sorting":"Namensverzeichnisse zu den Geburtsregistern des Standesamtes Pirna",
        "place_tsi":["Pirna"],
        "place_sorting":"Pirna",
        "type_usu":["multivolume_work"],
        "date_usi":["1876"],
        "date_sorting":"1876",
        "collection_tsi":["Stadtarchiv Pirna"],
        "collection_sorting":"Stadtarchiv Pirna",
        "volume":"",
        "timestamp":"2024-01-30T08:47:32.844Z"}]
  }}
```

**Comment by beatrycze-volk** [comment]
In fact it is correct and works as intended. This command removes only single document, not the parent document. Removing of the parent document would not be feasible as it can have also other children documents.

`https://digital.slub-dresden.de/data/kitodo/NamezuGeS_187317411X_0002/NamezuGeS_187317411X_0002_mets.xml` has also parent document and this one is put to database and index. Because of that at first you have 2 documents, then `delete` removes the child but parent stays, so at the second picture you have only one document. Also your index after `delete` contain only a parent document.

To remove the parent you would need to run `delete` command for parent document.

**Review by sebastian-meyer — APPROVED** [review]

**Pull request merged**